src/lasp_dependence_dag.erl

```diff
@@ -214,14 +214,10 @@ handle_call({will_form_cycle, From, To}, _From, #state{dag=Dag}=State) ->
 handle_call({add_edges, Src, Dst, Pid, ReadFuns, TransFun, {Dst, WriteFun}},
             _From, #state{dag=Dag, process_map=Pm}=State) ->
 
-    %% All other vertices are tracked at the `declare` level, but
-    %% stream is a special case. We check if it is already a vertex
-    %% in the graph since adding an already existing vertex removes
-    %% any previous vertex labels.
-    case (Dst =:= stream) andalso (digraph:vertex(Dag, Dst) =:= false) of
-        true -> digraph:add_vertex(Dag, Dst);
-        _ -> ok
-    end,
+    %% Add vertices only if they are either top or bottom. (See add_if)
```

Comment:
It just means vertices that can't have parents (top) or children (bottom).
A better name for top is source vertex and sink for bottom, although I don't know